include/aspect/utilities.h

```diff
@@ -110,6 +114,14 @@ namespace aspect
      * @param[in] property_name A name that identifies the type of property
      *   that is being parsed by this function and that is used in generating
      *   error messages if the map does not conform to the expected format.
+     * @param [in] allow_subentries If true allow having multiple subentries
+     *   for each key. If false only allow a single entry per key. In either
+     *   case each key is only allowed to appear once.
+     * @param [out] subentry_structure If handed over this
+     *   vector will be filled with as many components as
+     *   keys (+1 if there is a background field). Each value represents
+     *   how many subentries were found for this key. The sum over all
+     *   entries is the length of the returned vector of values.
      *
      * @return A vector of values that are parsed from the map, provided
      *   in the order in which the keys appear in the @p list_of_keys argument.
```

Comment:
You should add some documentation here that explains the new structure with subentries. That could also be a good place to explain why the `subentry_structure` is useful (it means whoever calls the function can figure out which value belongs to which key).